src/HttpPsrClient.php

```diff
@@ -30,9 +31,10 @@ public function sendRequest(RequestInterface $request, array $options = []);
      * @param RequestInterface[] $requests
      * @param array              $options
      *
-     * @return ResponseInterface[]
+     * @return BatchResult
      *
      * @throws Exception
+     * @throws BatchException
```

Comment:
add a description: "If one or more requests led to an exception, the BatchException is thrown. The BatchException also gives access to the BatchResult for the successful responses."